### PR TITLE
add Net5.0 to Test Infra and Bask SDK Tests

### PR DESCRIPTION
**Pull request by TimothyMothra**
This PR introduces net5.0 to our build infra, in preparation for #2190.

## Changes
- added Net5.0 to Microsoft.ApplicationInsights.Test.csproj
- Updated linux yaml
- Updated ADO build definition
- Disabled NetCoreApp2.1 in the Base Linux build. #2251 

### Checklist
- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Comment by TimothyMothra** [comment]
An unrelated test is failing. I'm investigating.

```
A total of 1 test files matched the specified pattern.
  Failed VerifyWhenRequestFailsWillWaitBeforeRetry [1 s]
  Error Message:
   Assert.IsTrue failed. 
  Stack Trace:
     at Microsoft.ApplicationInsights.Extensibility.Implementation.ApplicationId.ApplicationInsightsApplicationIdProviderTests.VerifyWhenRequestFailsWillWaitBeforeRetry() in /home/vsts/work/1/s/BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Extensibility/Implementation/ApplicationId/ApplicationInsightsApplicationIdProviderTests.cs:line 146
```

File: BASE/.vsts/linux-build.yml [hunk 1]
```diff
@@ -20,6 +22,11 @@ steps:
   displayName: install dotnet core 3.1
   inputs:
     version: 3.1.x
+
+- task: UseDotNet@2
```
Comment (cijothomas) [inline BASE/.vsts/linux-build.yml:26]: nit: i think we dont need to explictly install any .NET versions at all, as these linux machines come preinstalled.

Comment (TimothyMothra) [reply BASE/.vsts/linux-build.yml:26]: i added this because I was getting an error from this build that Net5.0 wasn't available.

**Pull request merged**